adapter/validator/src/test/java/org/mobilitydata/gtfsvalidator/validator/GtfsTypeValidatorTest.java

```diff
@@ -121,23 +119,116 @@ void outOfRangeFloatGenerateNotice() {
+                        "min:0.0 max:" + Float.MAX_VALUE + " actual:-66.7",
                 notice.getDescription());
 
         verify(mockFloatValidator, times(1)).isInRange(
-                ArgumentMatchers.eq(Float.valueOf(66.7f)),
-                ArgumentMatchers.eq(-6.66f),
-                ArgumentMatchers.eq(66.6f)
+                ArgumentMatchers.eq(Float.valueOf(-66.7f)),
+                ArgumentMatchers.eq(0f),
+                ArgumentMatchers.eq(Float.MAX_VALUE)
         );
     }
 
     @Test
-    void inRangeIntegerDoNotGenerateNotice() {
+    void negativeIntegerGenerateNotice() {
+        IntegerValidator mockIntegerValidator = mock(IntegerValidator.class);
+
+        GtfsSpecificationProto.CsvSpecProto mockFileSpec = mock(GtfsSpecificationProto.CsvSpecProto.class);
+        GtfsSpecificationProto.ColumnSpecProto mockColumnSpec = mock(GtfsSpecificationProto.ColumnSpecProto.class);
+        when(mockColumnSpec.getName()).thenReturn("non_negative_integer");
+
+        GtfsSpecificationProto.ColumnInputType mockInputType = mock(GtfsSpecificationProto.ColumnInputType.class);
+        when(mockInputType.getType()).thenReturn(GtfsSpecificationProto.ColumnInputType.InputType.NON_NEGATIVE_INTEGER);
+
+        when(mockColumnSpec.getType()).thenReturn(mockInputType);
+        when(mockFileSpec.getColumnList()).thenReturn(List.of(mockColumnSpec));
+
+        GtfsTypeValidator underTest = new GtfsTypeValidator(mockFileSpec,
+                mock(FloatValidator.class),
+                mockIntegerValidator,
+                mock(UrlValidator.class),
+                mock(Bcp47Validator.class),
+                mock(EmailValidator.class),
+                mock(RegexValidator.class),
+                Collections.emptySet()
+        );
+
+        Collection<Notice> result = underTest.validate(new ParsedEntity(
+                TEST_ID,
+                Map.of("non_negative_integer", -66),
+                new RawFileInfo.RawFileInfoBuilder().filename(TEST_FILE_TST).build()
+        ));
+
+        assertEquals(1, result.size());
+
+        Notice notice = new ArrayList<>(result).get(0);
+        assertThat(notice, instanceOf(IntegerFieldValueOutOfRangeNotice.class));
+        assertEquals("ERROR", ((IntegerFieldValueOutOfRangeNotice) notice).getLevel());
+        assertEquals(10, notice.getCode());
+        assertEquals("Out of range integer value", notice.getTitle());
+        assertEquals(TEST_FILE_TST, notice.getFilename());
+        assertEquals("Invalid value for field:`non_negative_integer` of entity with id:`test_id` -- " +
+                        "min:0 max:" + Integer.MAX_VALUE + " actual:-66",
+                notice.getDescription());
+
+        verify(mockIntegerValidator, times(1)).isInRange(
+                ArgumentMatchers.eq(Integer.valueOf(-66)),
+                ArgumentMatchers.eq(0),
+                ArgumentMatchers.eq(Integer.MAX_VALUE)
+        );
+    }
+
+    @Test
+    void zeroAsIntegerGenerateNotice() {
```

Comment:
The relationship between zero and Integer should be reworked. The type is `NON_NULL_INTEGER`. This is confusing